source/change-streams/change-streams.rst

```diff
@@ -483,27 +483,8 @@ Once a ``ChangeStream`` has encountered a resumable error, it MUST attempt to re
 
 - Perform server selection.
 - Connect to selected server.
-- If the ``ChangeStream`` has a cached ``postBatchResumeToken`` and has returned all documents in the most recent batch (or the most recent batch was empty):
-
-  - The driver MUST set ``resumeAfter`` to the cached ``postBatchResumeToken``.
-  - The driver MUST NOT set ``startAfter``. If ``startAfter`` was in the original aggregation command, the driver MUST remove it.
-  - The driver MUST NOT set ``startAtOperationTime``. If ``startAtOperationTime`` was in the original aggregation command, the driver MUST remove it.
-- Else if the ``ChangeStream`` has a cached ``documentResumeToken``:
-
-  - The driver MUST set ``resumeAfter`` to the cached ``documentResumeToken``.
-  - The driver MUST NOT set ``startAfter``. If ``startAfter`` was in the original aggregation command, the driver MUST remove it.
-  - The driver MUST NOT set ``startAtOperationTime``. If ``startAtOperationTime`` was in the original aggregation command, the driver MUST remove it.
-- Else if the ``ChangeStream`` was created with ``startAfter``:
-
-  - The driver MUST set ``resumeAfter`` to the value of the originally used ``startAfter``.
-  - The driver MUST NOT set ``startAfter``. The driver MUST remove it from the original aggregation command.
-  - The driver MUST NOT set ``startAtOperationTime``.
-- Else if the ``ChangeStream`` was created with ``resumeAfter``:
-
-  - The driver MUST set ``resumeAfter`` to the value of the originally used ``resumeAfter``.
-  - The driver MUST NOT set ``startAfter``.
-  - The driver MUST NOT set ``startAtOperationTime``.
-- Else if the ``ChangeStream`` has a saved operation time (either from an originally specified ``startAtOperationTime`` or saved from the original aggregation) and the max wire version is >= ``7``:
+- Call ``getResumeToken()`` to receive the resume token.
+- If the ``ChangeStream`` has a saved operation time (either from an originally specified ``startAtOperationTime`` or saved from the original aggregation) and the max wire version is >= ``7``:
```

Comment:
With this change, it looks as if a saved operation time always takes precedence over the resume token. In the original text, this was an "else if" below all of the resume token conditions and before the final "else" condition that re-sends the original aggregation w/o any options.

I think you need an initial "if" condition above this case to resume with the `resumeAfter` option set to the return value of `getResumeToken()` and only fall over to this case if `getResumeToken()` could not return anything.

Reply:
Also, I'd prefer not saying "Call getResumeToken()". [Exposing All Resume Tokens](https://github.com/mongodb/specifications/blob/master/source/change-streams/change-streams.rst#exposing-all-resume-tokens) gives drivers the option of implementing `getResumeToken()` or adding a new event.

Considering Jeremy's other comment, I'm thinking something like:
- Follow the rules of `getResumeToken()` to check for a resume token. If the rules apply:
   - The driver MUST set ``resumeAfter`` to what the rules return.
   - The driver MUST NOT set ``startAfter``. If ``startAfter`` was in the original aggregation command, the driver MUST remove it.
   - The driver MUST NOT set ``startAtOperationTime``. If ``startAtOperationTime`` was in the original aggregation command, the driver MUST remove it.
- otherwise, if the change stream has a saved operation time ...